bson/value_reader.go

```diff
@@ -730,11 +717,11 @@ func (vr *valueReader) ReadValue() (ValueReader, error) {
 			return nil, vr.invalidDocumentLengthError()
 		}
 
-		vr.pop()
+		_ = vr.pop()
```

Comment:
We should handle this error.

Reply:
This line is inside an error-handling block. We don't need to overwrite the original error even when `pop()` returns an error.

Reply:
See comment above.